lib/linguist/generated.rb

```diff
@@ -330,7 +330,8 @@ def go_vendor?
     #
     # Returns true or false.
     def npm_shrinkwrap?
-      !!name.match(/npm-shrinkwrap\.json/)
+      return false unless name.match(/npm-shrinkwrap\.json/)
+      !!name.match(/package-lock\.json/)
```

Comment:
Oh, and the first line should be `return true if name.match(/npm-shrinkwrap\.json/)`, otherwise it's equivalent to a logic AND. You could also write `name.match(/package-lock\.json/) || name.match(/npm-shrinkwrap\.json/)` by the way.